packages/amplify-category-api/src/category-utils/schema-reader.ts

```diff
@@ -0,0 +1,96 @@
+import * as fs from 'fs-extra';
+import path from 'path';
+import {
+  DocumentNode,
+  parse,
+} from 'graphql';
+import {
+  $TSAny,
+  $TSContext,
+  ApiCategoryFacade,
+} from 'amplify-cli-core';
+import { buildGraphQLTransformV2 } from '../graphql-transformer/transformer-factory';
+import {
+  SCHEMA_DIR_NAME,
+  SCHEMA_FILENAME,
+} from '../graphql-transformer/constants';
+import { generateTransformerOptions } from '../graphql-transformer/transformer-options-v2';
+import {
+  contextUtil,
+} from './context-util';
+
+/**
+ * SchemaReader is a utility point to consolidate and abstract GraphQL Schema reading
+ * The readSchema method provides a flag to read the un-processed (original) schema
+ * if desired, but by default the intent of the SchemaReader is to use the preProcess
+ * utility of the V2 transformer
+ */
+export class SchemaReader {
```

Comment:
Does this class need to be exported if the const is down below?

Reply:
Strictly speaking maybe not, but this is just how I've seen the singleton pattern done in TS in general, as well as some other points in the CLI. I think exporting the class as well as the instance might be helpful for linting